src/test/java/org/folio/rest/impl/DataExportTest.java

```diff
@@ -152,23 +165,54 @@ void testExport_GenerateRecordsOnFly(VertxTestContext context) throws IOExceptio
         .onSuccess(optionalJobExecution -> {
           JobExecution jobExecution = optionalJobExecution.get();
           fileDefinitionDao.getById(fileExportDefinitionCaptor.getValue().getId(), tenantId).onSuccess(optionalFileDefinition -> {
-              context.verify(() -> {
-                assertJobExecution(jobExecution, COMPLETED, EXPORTED_RECORDS_NUMBER_1);
-                validateExternalCallsForInventory();
-                context.completeNow();
-              });
+            context.verify(() -> {
+              assertJobExecution(jobExecution, COMPLETED, EXPORTED_RECORDS_NUMBER_1);
+              validateExternalCallsForInventory();
+              context.completeNow();
             });
+          });
+        });
+    });
+  }
+
+  @Test
+  void testExportByCQL_GenerateRecordsOnFly_andUnderlyingSrs(VertxTestContext context) throws IOException {
+    // given
+    String tenantId = okapiConnectionParams.getTenantId();
+    JsonObject bulkUUIDs = new JsonObject().put("ids", new JsonArray()
+      .add("7fbd5d84-62d1-44c6-9c45-6cb173998bbd")
+      .add("5fc04e92-70dd-46b8-97ea-194015762a60"));
+    Mockito.doReturn(Optional.of(bulkUUIDs)).when(mockInventoryClient).getInstancesBulkUUIDs(anyString(), any(OkapiConnectionParams.class));
```

Comment:
Why do you need to mock this? You already have the call specified in mock server, that should be able to return the values

Reply:
We don't use MockServer in this test

Reply:
We do, it extends the "RestVerticleTestBase" which automatically inititates the mock server. 
Also if you look at validateExternalCalls() , we validate the calls that are inititaed from mock server
This class is for testing end-end, so the actual idea is not to mock the calls here but rather have it tested end-end